src/components/InfiniteScroll/index.tsx

```diff
@@ -1,39 +1,56 @@
-import { Loader } from '@gnosis.pm/safe-react-components'
-import React, { ReactElement } from 'react'
-import { default as ReactInfiniteScroll, Props as ReactInfiniteScrollProps } from 'react-infinite-scroll-component'
-import styled from 'styled-components'
-
-import { Overwrite } from 'src/types/helpers'
-
-export const Centered = styled.div<{ padding?: number }>`
-  width: 100%;
-  height: 100%;
-  display: flex;
-  padding: ${({ padding }) => `${padding}px`};
-  justify-content: center;
-  align-items: center;
-`
-
-export const SCROLLABLE_TARGET_ID = 'scrollableDiv'
-
-type InfiniteScrollProps = Overwrite<ReactInfiniteScrollProps, { loader?: ReactInfiniteScrollProps['loader'] }>
-
-export const InfiniteScroll = ({ dataLength, next, hasMore, ...props }: InfiniteScrollProps): ReactElement => {
-  return (
-    <ReactInfiniteScroll
-      style={{ overflow: 'hidden' }}
-      dataLength={dataLength}
-      next={next}
-      hasMore={hasMore}
-      loader={
-        <Centered>
-          <Loader size="md" />
-        </Centered>
-      }
-      scrollThreshold="120px"
-      scrollableTarget={SCROLLABLE_TARGET_ID}
-    >
-      {props.children}
-    </ReactInfiniteScroll>
-  )
+import React, {
+  createContext,
+  Dispatch,
+  forwardRef,
+  ReactElement,
+  ReactNode,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react'
+import { useInView } from 'react-intersection-observer'
+
+export const INFINITE_SCROLL_CONTAINER = 'infinite-scroll-container'
+
+export const InfiniteScrollContext = createContext<{
+  ref?: any
+  lastItemId?: string
+  setLastItemId: Dispatch<SetStateAction<string>>
+}>({ setLastItemId: () => {} })
+
+export const InfiniteScrollProvider = forwardRef<HTMLDivElement, { children: ReactNode }>(
+  ({ children }, ref): ReactElement => {
+    const [lastItemId, setLastItemId] = useState<string>()
```

Comment:
shouldn't it be string | undefined?

Reply:
no, `undefined` is implicit

![image](https://user-images.githubusercontent.com/3315606/108514993-57034580-72a3-11eb-94fd-11d66cef6146.png)